src/node_file.cc

```diff
@@ -844,13 +844,13 @@ void AfterOpenFileHandle(uv_fs_t* req) {
 
 // Reverse the logic applied by path.toNamespacedPath() to create a
 // namespace-prefixed path.
-void FromNamespacedPath(std::string* path) {
+void FromNamespacedPath(std::string& path) {  // NOLINT(runtime/references)
```

Comment:
Why is the pointer being changed to a non-const reference? This is going against our style guide - https://github.com/nodejs/node/blob/895cc572ac9dee8287a2365ca0087940070ff632/doc/contributing/cpp-style-guide.md#avoid-non-const-references, right?

Reply:
This function calls a std::string function without checking if the pointer is nullptr. I made this change to make it more safe, and avoid passing nullptr. I believe the cpplint does not apply here. If it does, I didn’t understand it.

Reply:
Oh, in that case, we could add a `DCHECK_NOT_NULL(path)`

Reply:
Makes sense. I'll update it. Thanks @RaisinTen